go-controller/pkg/generator/ip_generator.go

```diff
@@ -8,21 +8,21 @@ import (
 	utilnet "k8s.io/utils/net"
 )
 
-// ipGenerator is used to generate an IP from the provided CIDR and the index.
+// IPGenerator is used to generate an IP from the provided CIDR and the index.
 // It is not an allocator and doesn't maintain any cache.
-type ipGenerator struct {
+type IPGenerator struct {
 	netCidr   *net.IPNet
 	netBaseIP *big.Int
 }
 
-// newIPGenerator returns an ipGenerator instance
-func newIPGenerator(subnet string) (*ipGenerator, error) {
+// NewIPGenerator returns an ipGenerator instance
+func NewIPGenerator(subnet string) (*IPGenerator, error) {
```

Comment:
We can call this just `New` so we do `ipgenerator.New` and we don't repeat ourselves.

Reply:
on second thoughts I think NewIPGenerator() is cleaner..

Reply:
Good enough